lib/src/sv_internal.h

```diff
@@ -457,20 +457,12 @@ port_settings_to_onvif(signed_video_t *self);
 /*
  * This function initializes ONVIF settings by porting all the settings,
  * retrieving the private key and certificate chain from the signed video object,
- * and setting the ONVIF signing key pair.
+ * and setting the ONVIF signing key pair. The function will free the ONVIF object
+ * on failure, allowing the framework to continue with the signed video.
```

Comment:
Yes, this is the desired behavior, but is that really what happens with your implementation?
Have you tested/verified that?